src/classification/main.jl

```diff
@@ -514,14 +514,85 @@ function build_forest(
     return _build_forest(forest, size(features, 2), n_trees, impurity_importance)
 end
 
+const ERR_CANT_UPDATE_IMPURITY_IMPORTANCE = DimensionMismatch(
+    "Looks like you want to add trees to a model previously trained using a "*
+        "different number of features, which means impurity importances "*
+        "cannot be updated. Fix by setting `impurity_importance=false`. "
+)
+
+"""
+    build_forest(model, labels, features, options...; keyword_options...)
+
+Return an updated version of `model` with additional `n_trees=options[2]` added to the
+forest. Here `options` and `keyword_options` are as for the regular `build_forest` method,
+which excludes the `model` argument.
+
+Even if training data is the same in all `build_forest` calls, it is not practically
+possible to guarantee adding trees in steps is identical to adding them all at once,
+because of the way random number generators are generated and used. But in all other
+respects these approaches are equivalent.
+
+# Example
+
+```
+features, labels = load_data("iris")
+features = float.(features)
+labels = string.(labels)
+```
+
+The call
+
+```
+model = build_forest(labels, features, 2, 200) # n_trees = 200
+```
+
+is approximately equivalent to
+
+```
+model1 = build_forest(labels, features, 2, 150) # n_trees = 150
+model = build_forest(model1, labels, features, 2, 50) # n_trees = 50
+```
+
+"""
+function build_forest(
+    model               :: Ensemble{S,T},
+    labels              :: AbstractVector{T},
+    features            :: AbstractMatrix{S},
+    options...;
+    impurity_importance=true,
+    kwoptions...,
+    ) where {S, T}
+
+    # Only compute impurity importances if requested and present in the existing ensemble:
+    impurity_importance = impurity_importance && has_impurity_importance(model)
+
+    # Combining forests will throw an error if feature importances are reqested now (and
+    # were stored previously) and if the number of features has also changed. So we catch
+    # that before training new ensemble:
+    n_features = size(features, 2)
+    !impurity_importance ||
+        n_features == DecisionTree.n_features(model) ||
+            throw(ERR_CANT_UPDATE_IMPURITY_IMPORTANCE)
```

Comment:
```suggestion
if (!impurity_importance || n_features == DecisionTree.n_features(model))
    throw(ERR_CANT_UPDATE_IMPURITY_IMPORTANCE)
end
```
Does this mean the same? I find the double OR hard to understand

Reply:
No, this is not the same. Rather the equivalent would be:

```
if impurity_importance && n_features != DecisionTree.n_features(model)
    throw(ERR_CANT_UPDATE_IMPURITY_IMPORTANCE)
end
```

I agree this is clearer and will change it.

Reply:
Done.

Reply:
Thanks